### ISSUE TITLE
Use snake_case for methods and variables

### ISSUE THREAD

**Issue by jihwans** [bbatsov/rubocop#1351]
There are cases there you want to have module method with the same name as a class.

Here's an example: - from Sequel project -

~~~
module Sequel
  def self.Model(source) # RuboCop Complains here..
   ...
  end

  class Model
  ...
  end
end
~~~

Here, Sequel::Model is a class emitter hence the name that resembles the name of the class.
The class definition with same name is included in the file.
I guess this should be allowed... ?

**Comment by hl** [comment]
I'm also getting this warning with the following code and think this should be allowed as well?

```ruby
module Admin
  Import = Admin::Container::Inject.args

  def self.Import(*args)
    Import[*args]
  end
end
```

**Comment by mikegee** [comment]
@hl I think your comment would be more appropriate on #2108 or as a new issue.

Looking at the implementation of #2108, I think it only allows method names matching classes in the same scope. I'm not sure if `Import` from your example is a class, or if RuboCop can know that it is a class since it is performing static analysis.

**Comment by jonas054** [comment]
It seems reasonable to allow capitalized method names for singleton methods that correspond to a constant in the same scope. Even if it's not a class definition.

But this issue is closed so please open a new one, @hl.
